### PR TITLE
Remove `generate_doc` feature

### PR DESCRIPTION
**Pull request by jprendes**
This PR removes the `generate_doc` feature.
The expectation is that updating `README.md` should be done directly.

### CODE REVIEW

File: crates/containerd-shim-wasm/doc/doc.md [hunk 1]
```diff
@@ -1,36 +0,0 @@
-A library to help build containerd shims for wasm workloads.
-
-## Usage
```
Comment (Mossaka) [inline crates/containerd-shim-wasm/doc/doc.md:3]: Why is the usage doc being removed?

Comment (jprendes) [reply crates/containerd-shim-wasm/doc/doc.md:3]: This is `Usage` is duplicated in `README.md`.
It was basically `cat doc/header.md doc/doc.md > README.md`
This PR proposes not using `doc/*.md`, and just work on `README.md` directly.

File: crates/containerd-shim-wasm/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-![runwasi logo](../../art/logo/runwasi_logo_icon.svg)
```
[the remark this answers is not in the document]
Comment (jprendes) [reply crates/containerd-shim-wasm/README.md:1]: There's no change in that respect.
The text sourced for the doc is exactly the same (modulo now including the logo).
Also this only concerns the `containerd-shim-wasm` crate's `README.md`, which is different from the project's `README.md` and doesn't have any relative links other than the image.

[the remark this answers is not in the document]
Comment (jsturtevant) [reply crates/containerd-shim-wasm/README.md:1]: ah I missed that it wasn't the projects readme and only the crate readme.

Looks like there is some duplication between the crate readme and the main readme that should probably be moved the to crate readme and referenced from the main.

**Review by Mossaka — APPROVED** [review]
LGTM

**Review by Mossaka — APPROVED** [review]

**Pull request merged**